innersource-license.md

```diff
@@ -0,0 +1,83 @@
+## Title
+
+InnerSource License
+
+## Patlet
+
+Two companies that belong to the same group want to share software source code with each other but they are concerned about the implications in terms of legal liabilities or cross-company accounting. An **InnerSource License** provides a reusable legal framework for the inter-company agreement within the group, which opens up new collaboration options.
```

Comment:
Cornelius and I also went back and forth between calling it *enterprise* or *group*. And we landed at *group* eventually.

For the time being I suggest to keep it as *group* so that we don't have to reword everything. Also I don't know which terms are easier to understand, or which ones are more correct in legal speak.

Maybe when other companies that have similar challenges add their experience to this pattern, then we can test which term they prefer?

Reply:
I really like the edits you did to the Patlet!
I will take parts of that but keep the group/company terms as they are for the time being.

Reply:
I was only able to work in part of that, without changing the original meaning of the Patlet too much.

Will keep this conversation open also for reference for Cornelius.

Reply:
> For the time being I suggest to keep it as group so that we don't have to reword everything.

I know writing is a lot of work and time is a scarce resource. Let's still try to find the best terminology possible. We / I can support with the rewording once we settled on terms.

> Maybe when other companies that have similar challenges add their experience to this pattern, then we can test which term they prefer?

In the organizations I worked with, we ended up using exclusively `legal entity` to talk about the - well - legal entities.

Reply:
Summary of arguments, both gathered from this discussion and from my head:

`group, company`
- [PLUS] Fits well with the proposed pattern (because it's the terms the author's had in mind initially)
- [PLUS] Less re-writing burden
- [MINUS] Excludes specific organizations that are not groups/companies but still consist of multiple legal entities and want to do InnerSource (private / publich research institutions, government agencies, ...)
- [MINUS] Ambiguity of both terms
- [MINUS] Less clear on describing the actual type of building block of the organization that is relevant for this pattern (which, as I read it, applies primarily to multiple legal entities)

Reply:
> So, my suggestion would still be, to use different terms, probably organization and legal entity

I like the term `legal entity` because that's closely tied to the need of having legal agreements in place. `organization` as a general term would also work for me. No matter what terms we use, we should make sure that we have a definition somewhere which explains them in the context of this pattern.

Reply:
> No matter what terms we use, we should make sure that we have a definition somewhere which explains them in the context of this pattern.

+1000! Let's add those definitions to the context section

Reply:
I'll go with @cornelius point of "what ever we use we should make sure that we have a definition somewhere". Let's use a glossary for that one. We argued that this would be useful across patterns in any case. Starting it can be as simple as creating a simple text file with "term: explanation" pairs in it and link to that, it's hypertext after all. 
We can always fancy that up later.

Reply:
Also see inline comment from @lenucksi:
https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/147/files#r417457368

Reply:
Based on all the feedback I will rewrite the pattern to use **organization** (for group/enterprise) and **legal entity** (for company/subsidiary). I will put those as definitions to the Context section.

Will resolve this thread here once done.

Reply:
I have rewritten the pattern accordingly.
 
I have placed the glossary at the end of this file, as I cannot claim that this is how these terms are used in other patterns. If we really want to produce a global glossary, I suggest that we do that as a separate activity outside of this PR>

@maxcapraro would you mind reviewing this, as you brought this topic up?

My observations after doing the rewrite:

- the pattern is certainly more consistent now 👍 
- it sounds much more like a legal document, and less like it is written for humans (which I personally don't like) 👎 
- there are a couple of places where replacing the word "company" without "legal entity" would make the sentences longer,  and also a bit awkward. I have left those as they were. Examples:
  - cross-company accounting
  - Company culture
  - inter-company relationships

Still I think that this was a good iteration and if this gets sign-off, I would gladly get this merged to master as it already provides value to the InnerSource community as is.